libs/sr-formula/src/data/util/sheet.ts

```diff
@@ -13,3 +30,105 @@ export function register(
     Array.isArray(data) ? data.map(internal) : internal(data)
   )
 }
+
+function registerFormula(
+  name: string,
+  team: boolean | undefined,
+  q: 'dmg' | 'heal' | 'shield',
+  cond: string | StrNode,
+  ...extra: TagMapNodeEntries
+): TagMapNodeEntries {
+  reader.name(name) // register name:<name>
+  const listing = (team ? teamBuff : selfBuff).listing.formulas
+  return [
+    listing.add(listingItem(reader.withTag({ name, qt: 'formula', q }), cond)),
+    ...extra.map(({ tag, value }) => ({ tag: { ...tag, name }, value })),
+  ]
+}
+
+export function listingItem(t: Read, cond?: string | StrNode) {
+  return tag(cond ?? t.ex ?? 'unique', t.tag)
+}
+
+/**
+ * Creates an array of TagMapNodeEntries representing a damage instance split by their multipliers, and registers their formulas
+ * @param name Base name to be used as the key
+ * @param elementalTypeKey Elemental type of the damage
+ * @param base Node representing the full damage value
+ * @param splits Array of decimals that should add up to 1. Each entry represents the percentage of damage that hit deals, for multi-hit moves. We get splits from SRSim devs, see the array at the top of https://github.com/simimpact/srsim/blob/main/internal/character/march7th/ult.go for example.
+ * @param arg `{ team: true }` to use `teamBuff` instead of `selfBuff`. `{ cond: <node> }` to hide these instances behind a conditional check.
```

Comment:
We could also say that `team:true` will include the formula in teammates' listing as well, while `team:false` only includes the formula in current character's listing